static/js/filter.js

```diff
@@ -99,6 +99,17 @@ function message_matches_search_term(message, operator, operand) {
     case 'sender':
         return people.id_matches_email_operand(message.sender_id, operand);
 
+    case 'huddle-with':
+        var operand_id = people.get_user_id(operand);
+        if (!operand_id) {
+            return false;
+        }
+        var message_ids = people.huddle_with_user_ids(message);
+        if (!message_ids) {
```

Comment:
this variable should be called `user_ids`.  `message_ids` sounds like a list of message IDs!  It's really important to choose variable names carefully to avoid making code confusing.